File: frontend/src/api/forecast.ts
import { apiRequest } from './client';

export interface ForecastPoint {
  step: number;
  value: number;
}

export interface ForecastRequest {
  symbol: string;
  historical_values: number[];
  horizon: number;
}

export interface ForecastResponse {
  symbol: string;
  horizon: number;
  points: ForecastPoint[];
}

export function createForecast(payload: ForecastRequest) {
  return apiRequest<ForecastResponse>('/forecast', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
}

File: backend/app/domain/entities/forecast.py
from dataclasses import dataclass


@dataclass(frozen=True)
class ForecastInput:
    symbol: str
    historical_values: list[float]
    horizon: int


@dataclass(frozen=True)
class ForecastPoint:
    step: int
    value: float


@dataclass(frozen=True)
class ForecastResult:
    symbol: str
    horizon: int
    points: list[ForecastPoint]
